src/server/lib/ab.ts

```diff
@@ -60,9 +72,83 @@ export const selectVariant = <V extends Variant, T extends Test<V>>(test: T, mvt
     return selectWithSeed(mvtId, seed, test.variants);
 };
 
-export const selectVariantNonSticky = <V extends Variant, T extends Test<V>>(test: T): V => {
-    const index = Math.floor(Math.random() * test.variants.length);
-    return test.variants[index];
+const selectVariantWithMethodology = <V extends Variant, T extends Test<V>>(
+    test: T,
+    mvtId: number,
+    banditData: BanditData[],
+    methodology: Methodology,
+): V | undefined => {
+    if (methodology.name === 'EpsilonGreedyBandit') {
+        return selectVariantUsingEpsilonGreedy(banditData, test, methodology.epsilon);
+    }
+    return selectVariantUsingMVT<V, T>(test, mvtId);
+};
+
+const addMethodologyToTestName = (testName: string, methodology: Methodology): string => {
+    if (methodology.name === 'EpsilonGreedyBandit') {
+        return `${testName}_EpsilonGreedyBandit-${methodology.epsilon}`;
+    } else {
+        return `${testName}_ABTest`;
+    }
+};
+
+/**
+ * Selects a variant from the test based on any configured methodologies.
+ * Defaults to an AB test.
+ */
+export const selectVariant = <V extends Variant, T extends Test<V>>(
+    test: T,
+    mvtId: number,
+    banditData: BanditData[],
+): { test: T; variant: V } | undefined => {
+    if (test.methodologies && test.methodologies.length === 1) {
+        // Only one configured methodology
+        const variant = selectVariantWithMethodology<V, T>(
+            test,
+            mvtId,
+            banditData,
+            test.methodologies[0],
+        );
+        if (variant) {
+            return {
+                test,
+                variant,
+            };
+        }
+    } else if (test.methodologies) {
+        // More than one methodology, pick one of them using the mvt value
+        const methodology =
+            test.methodologies[getRandomNumber(test.name, mvtId) % test.methodologies.length];
+
+        // Add the methodology to the test name so that we can track them separately
+        const testWithNameExtension = {
+            ...test,
+            name: addMethodologyToTestName(test.name, methodology),
+        };
+        const variant = selectVariantWithMethodology<V, T>(
+            testWithNameExtension,
+            mvtId,
+            banditData,
+            methodology,
+        );
+        if (variant) {
+            return {
+                test: testWithNameExtension,
+                variant,
+            };
+        }
+    } else {
+        // No configured methodology, default to AB test
+        const methodology: Methodology = { name: 'ABTest' };
+        const variant = selectVariantWithMethodology<V, T>(test, mvtId, banditData, methodology);
```

Comment:
Could call `selectVariantUsingMVT` instead? Had to double check but don't think specifying 'ABTest' is doing anything here

Reply:
good point!